Added documentation for `Divider` component (#2225)
iTwin/iTwinUI: apps/website/src/content/docs/divider.mdx

---
title: Divider
description: A divider can be used to separate a layout into different sections of content or element.
propsPath: '@itwin/itwinui-react/esm/core/Divider/Divider.d.ts'

> ---

Reviewer: We should put this component under Utilities in the sidebar. Also we should add a thumbnail in the future. For now lets use a TODO comment.

```suggestion
group: utilities
thumbnail: #TODO
---
```

Author: I have added this component under Utilities. Just a quick question, how are thumbnails generated? I noticed some components do have thumbnails while some remain "#TODO".

Author: I see, thank you.

<p>{frontmatter.description}</p>

<LiveExample src='Divider.main.jsx' truncate={false}>
  <AllExamples.DividerMainExample client:load />
</LiveExample>

> ## Usage
> 
> <p>
>   The divider is set to be displayed both horizontally (by default) or vertically. Additionally, the
>   divider can be customized to suit different design needs, including adjustable styles such width
>   or height.
> </p>

Reviewer: This Usage section feels superfluous and disconnected from the vertical example. Maybe we can skip it and instead create a section for "Vertical". Also, the `<p>` tags are not necessary because this is markdown.

```suggestion
## Vertical

The `Divider` component is displayed horizontally by default. To display it vertically, use the `orientation` prop.
```

Author: I have reworded this section.

<LiveExample src='Divider.vertical.jsx' truncate={false}>
  <AllExamples.DividerVerticalExample client:load />
</LiveExample>

## Props

> <PropsTable path={frontmatter.propsPath} />

Reviewer: The frontmatter variable is not necessary, we can use the string directly.

```suggestion
<PropsTable path='@itwin/itwinui-react/esm/core/Divider/Divider.d.ts' />
```

We should actually remove these `propsPath` variables from all places (even in other components). Feel free to open a separate PR for that.

Author: I will add this as an after-PR TODO and will commit the suggested change in this PR.

Author: I have made the change.
